### PR TITLE
Introduce limit range to match both data and LUT ranges

### PR DESCRIPTION
**Pull request by LucasGandel**

### CODE REVIEW

**Comment by LucasGandel** [comment]
@finetjul 
Here are the range related changes. It is also introducing the accessor on the discretizable color transfer function

File: Libs/Visualization/VTK/Core/vtkScalarsToColorsContextItem.cpp [hunk 1]
```diff
@@ -201,6 +213,26 @@ void vtkScalarsToColorsContextItem::SetHistogramTable(vtkTable* table,
...
+  limitRange[1] = std::max(ctfRange[1], max);
+
+  this->EditorChart->GetAxis(vtkAxis::BOTTOM)->SetUnscaledRange(limitRange);
+  this->EditorChart->RecalculateBounds();
+
+  this->HistogramChart->GetAxis(vtkAxis::BOTTOM)->SetUnscaledRange(limitRange);
+  this->HistogramChart->RecalculateBounds();
+
+  this->PreviewChart->GetAxis(vtkAxis::BOTTOM)->SetUnscaledRange(limitRange);
+  this->PreviewChart->RecalculateBounds();
```
Comment (finetjul) [inline Libs/Visualization/VTK/Core/vtkScalarsToColorsContextItem.cpp]: please factorize with vtkScalarsToColorsContextItem::SetDiscretizableColorTransferFunction

-> e.g. both methods call a unique method

File: Libs/Visualization/VTK/Widgets/ctkVTKDiscretizableColorTransferWidget.cpp [hunk 2]
```diff
@@ -353,14 +353,17 @@ void ctkVTKDiscretizableColorTransferWidget::setColorTransferFunction(
...
-
   d->previousOpacityValue = 1.0;
   d->opacitySlider->setValue(d->previousOpacityValue);
 
+  double* crng = d->scalarsToColorsContextItem->
+    GetDiscretizableColorTransferFunction()->GetRange();
+  double limitRange[2];
+  limitRange[0] = std::min(crng[0], d->dataRange[0]);
+  limitRange[1] = std::max(crng[1], d->dataRange[1]);
+  d->rangeSlider->setRange(limitRange[0], limitRange[1]);
```
Comment (finetjul) [inline Libs/Visualization/VTK/Widgets/ctkVTKDiscretizableColorTransferWidget.cpp]: can't you get the limit from the chart ?

File: Libs/Visualization/VTK/Widgets/ctkVTKDiscretizableColorTransferWidget.cpp [hunk 3]
```diff
@@ -416,6 +428,18 @@ void ctkVTKDiscretizableColorTransferWidget::setHistogram(
...
+  {
+    return;
+  }
+
+  double* crng = d->scalarsToColorsContextItem->
+    GetDiscretizableColorTransferFunction()->GetRange();
+  double limitRange[2];
+  limitRange[0] = std::min(crng[0], d->dataRange[0]);
+  limitRange[1] = std::max(crng[1], d->dataRange[1]);
+  d->rangeSlider->setRange(limitRange[0], limitRange[1]);
```
Comment (finetjul) [inline Libs/Visualization/VTK/Widgets/ctkVTKDiscretizableColorTransferWidget.cpp]: same here, that min/max code should be written only once

File: Libs/Visualization/VTK/Widgets/ctkVTKDiscretizableColorTransferWidget.cpp [hunk 4]
```diff
@@ -416,6 +428,18 @@ void ctkVTKDiscretizableColorTransferWidget::setHistogram(
...
+
   d->ScalarsToColorsView->GetInteractor()->Render();
```
Comment (finetjul) [inline Libs/Visualization/VTK/Widgets/ctkVTKDiscretizableColorTransferWidget.cpp]: is it really needed ?

File: Libs/Visualization/VTK/Core/vtkScalarsToColorsContextItem.cpp [hunk 5]
```diff
@@ -201,6 +203,13 @@ void vtkScalarsToColorsContextItem::SetHistogramTable(vtkTable* table,
 void vtkScalarsToColorsContextItem::SetDataRange(double min, double max)
 {
   this->EditorChart->SetDataRange(min, max);
+
+  if (this->GetDiscretizableColorTransferFunction() == nullptr)
+  {
+    return;
+  }
```
Comment (finetjul) [inline Libs/Visualization/VTK/Core/vtkScalarsToColorsContextItem.cpp]: must go in RecalculateChartsRange()

File: Libs/Visualization/VTK/Widgets/ctkVTKDiscretizableColorTransferWidget.cpp [hunk 6]
```diff
@@ -376,6 +383,15 @@ ctkVTKDiscretizableColorTransferWidget::colorTransferFunction() const
 }
 
 // ----------------------------------------------------------------------------
+vtkDiscretizableColorTransferFunction*
+ctkVTKDiscretizableColorTransferWidget::discretizableColorTransferFunction()
+const
```
Comment (finetjul) [inline Libs/Visualization/VTK/Widgets/ctkVTKDiscretizableColorTransferWidget.cpp]:
```vtkDiscretizableColorTransferFunction* ctkVTKDiscretizableColorTransferWidget
::discretizableColorTransferFunction() const```

**Pull request merged**